shared-lib/lib/Expression/ExpressionFunctions.ts

```diff
@@ -3,6 +3,23 @@ import { JSONPath } from 'jsonpath-plus'
 
 // Note: when adding new functions, make sure to update the docs!
 export const ExpressionFunctions: Record<string, (...args: any[]) => any> = {
+	// General operations
+	length: (v) => {
+		let len = 0
+		if (v === undefined || v === null) {
+			len = 0
+		} else if (Array.isArray(v)) {
+			len = v.length
+		} else if (typeof v === 'object') {
+			len = Object.keys(v).length
+		} else if (typeof v === 'number') {
+			len = (v + '').length
+		} else {
```

Comment:
Not all uncatched types have a lenght property, e.g. BIGINT

Reply:
I've added BigInt. What others are you aware (RegExp maybe) of/what do you think I should do in the default case?

Reply:
Added RegExp too now...

Reply:
Hm, I think string should have its own case and then it is not part of the unknown rest any more. The unknown rest could maybe return NaN.
From the JS perspective there are also iterators and generators that are interesting for length() but currently we don't use them in variables, so if you want to implement them it would be a bonus for possible future use. Also we don't use regex data type currently and I'm not sure about the value of knowing the length of a regex.

Reply:
yeah I agree, a string case would be good and the fallback should always return NaN.

Another note is that `typeof new Regexp()` is `object`, so this will need to be above the object check

Reply:
Okay I've added/sorted those others, useful or not and made it return NaN as the default otherwise.

I guess ideally we need yet another type of object if we wish to test the NaN case...